File: longest-palindrome-string/LongestPalindromeString.cs
using System;

public class LongestPalindromeString 
{
    public static string LongestPalindromeStrings(string str) {
       if(str.Length == 1) { return str; }
        string longestSubstring = "";
        for(var i = 0; i < str.Length; i++) {
            for(var j = i; j < str.Length; j++) {
                string substring = str.Substring(i, j + 1 - i);
                if(substring.Length > longestSubstring.Length && IsPalindrome(substring)) {
                    longestSubstring = substring;
                }
            }
        }
        return longestSubstring;
    }

    public static bool IsPalindrome(string str) {
        for(var i = 0; i < str.Length / 2; i++) {
            if(Char.ToLower(str[i]) != Char.ToLower(str[str.Length - i - 1])) return false;
        }
        return true;
    }
}

File: longest-palindrome-string/LongestPalindromeString.swift
import Foundation

public class LongestPalindromeString {
    func longestPalindromicSubstring(string: String) -> String {
        if(string.count == 1) { return string }
            var longestPalindrome = ""
            for i in 0 ..< string.count {
                for j in i ..< string.count {
                    let leftIndex = string.index(string.startIndex, offsetBy: i)
                    let rightIndex = string.index(string.startIndex, offsetBy: j)
                    let substring = String(string[leftIndex ... rightIndex])
                    if substring.count > longestPalindrome.count, isPalindrome(substring) {
                        longestPalindrome = substring
                    }
                }
            }
            return longestPalindrome
    }

    func isPalindrome(_ str: String) -> Bool {
            let characters: [Character] = Array(str)
            for i in stride(from: 0, to: characters.count / 2, by: 1) {
                if(characters[i].lowercased() != characters[characters.count - i - 1].lowercased()) {
                    return false
                }
            }
            return true
    }
}